geth-with-protocol/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ethereum/client-go:v1.9.0 as builder-geth
+FROM ethereum/client-go:v1.9.3 as builder-geth
```

Comment:
If we're upgrading the geth version, perhaps we can just use the most recent version?

Reply:
invalid v,r,s values . Could use latest indeed, I chose to match whatever client version we use in go-livepeer.

Reply:
> invalid v,r,s values

Hm I don't think there was anything in the Istanbul upgrade that would cause signature validation behavior to change at an EVM level. The [OpenZepplin signature library](https://github.com/livepeer/protocol/blob/d6bcb03337839071f1a90e391760ef1f7660f836/contracts/pm/mixins/MixinTicketBrokerCore.sol#L9) is where v, r, s checking happens. Unless I'm missing something here, I suspect that the geth version may not be the problem.

> I chose to match whatever client version we use in go-livepeer.

Gotcha. We actually don't need to match the version of geth libraries used within go-livepeer so I suggest we just use the latest to pull in any fixes/improvements included in more recent releases.

Reply:
For the record, I was having issues syncing 1.9.10 but 1.9.11 was a pleasure.  I can confirm that it runs well & stable.

Reply:
Logs 

```
> protocol@1.0.0 migrate /psrc/protocol
> truffle migrate "--network=lpTestNet"


Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.



Starting migrations...
======================
> Network name:    'lpTestNet'
> Network id:      54321
> Block gas limit: 0x8000000


1_initial_migration.js
======================
WARN [02-25|17:57:24.535] Served eth_estimateGas                   conn=127.0.0.1:49362 reqid=10 t=21.6927ms err="gas required exceeds allowance (8000000) or always failing transaction"

   Deploying 'Migrations'
   ----------------------
WARN [02-25|17:57:26.006] Served eth_sendTransaction               conn=127.0.0.1:49370 reqid=14 t=18.7691ms err="invalid transaction v, r, s values"

Error: Error: Error:  *** Deployment Failed ***

"Migrations" received a generic error from Geth that
can be caused by hitting revert in a contract constructor or running out of gas.
   * Returned error: gas required exceeds allowance (8000000) or always failing transaction.
   * Try: + using the '--dry-run' option to reproduce this failure with clearer errors.
          + verifying that your gas is adequate for this deployment.

    at Object.run (/psrc/protocol/node_modules/truffle/build/webpack:/packages/migrate/index.js:96:1)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Truffle v5.0.43 (core: 5.0.43)
Node v10.19.0
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! protocol@1.0.0 migrate: `truffle migrate "--network=lpTestNet"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the protocol@1.0.0 migrate script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-02-25T17_57_26_124Z-debug.log
```

Reply:
hmm sorry, bumping the version doesn't actually change anything for the error.

I forgot that for building my own image I had also changed the hard fork block numbers in  the genesis file although we'd likely have to only change the EIP155 blocknumber to be earlier. 

This might only occur using 5 sec blocktimes 

https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md

Reply:
> This might only occur using 5 sec blocktimes

Yeah this likely only occurs when the contracts are deployed before the 3rd block (the current EIP155 block number specified in the [genesis file](https://github.com/livepeer/docker-livepeer/blob/671af73b21e54e4f9037e5ee422fea7b088b484b/geth-with-protocol/genesis.json#L7)) is mined. So, with 5 sec block times, if the contract deployment starts before 15 sec have elapsed then we hit the transaction signature error during deployment. If we want to avoid that issue the hard fork block numbers could be set to 0.

Reply:
I think it needs to be 1, with 0 the HF isn't implemented iirc.

Reply:
Have you checked if setting the HF block numbers to 0 works? I would expect 0 to be a valid value for the HF block number fields in the genesis file. Geth should just set the appropriate consensus rules at the specified block number. So, 0 would just mean that all the latest consensus rules are activated starting from the genesis block.

Reply:
fixed 2ccfcc0